src/plugins/iconv/iconv.c

```diff
@@ -186,11 +186,13 @@ static int validateFile(Plugin *handle, Key *parentKey)
 {
 	const char *fileName = keyString(parentKey);
 	FILE *fp = fopen(fileName, "rb");
-	char inputBuffer[4096];
+	char *inputBuffer = NULL;
+	const int inputBufferSize = 4096;
 	unsigned long counter = 0;
 	while(!feof(fp))
 	{
-		size_t bytesRead = fread(inputBuffer, CHAR_SIZE_MAX, (sizeof(inputBuffer)/CHAR_SIZE_MAX), fp);
+		inputBuffer = realloc(inputBuffer, inputBufferSize);
```

Comment:
can you use elektraRealloc here? Why is it reallocated every time? inputBufferSize is not changed?